operator/controllers/kyma_controller.go

```diff
@@ -416,6 +419,15 @@ func (r *KymaReconciler) SetupWithManager(setupLog logr.Logger, mgr ctrl.Manager
 		handler.EnqueueRequestsFromMapFunc(r.TemplateChangeHandler().Watch(context.TODO())),
 		builder.WithPredicates(predicate.GenerationChangedPredicate{}))
 
+	controllerBuilder = controllerBuilder.Watches(
```

Comment:
We need to check if this does not trigger the `reconcile` function. Since the Builder only works if the `forInput` is not empty. This input is set in line 379. Then if an event happens, the respective handler function gets triggered and then the reconcile function. We are currently handling the events in the same way in the watcher, and there we encounter exactly this problem.

Reply:
Concern resolved after we tested it together 😄